hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+        # and not 200
+        return Response(data={'resource_id': pk}, status=status.HTTP_200_OK)
+
+
+class ResourceList(generics.ListAPIView):
+    """
+    Get a list of resources based on the following filter query parameters
+
+    For an anonymous user, all public resources will be listed.
+    For any authenticated user with no other query parameters provided in the request, all resources that are viewable
+    by the user will be listed.
+
+    REST URL: hsapi/resource-list/{query parameters}
+
+    HTTP method: GET
+
+    HTTP status code (on success):200
+
+    Supported query parameters:
+
+    :type   owner: str
+
+    :type   types: list of resource type class names
+
+    :type   from_date:  str (e.g., 2015-04-01)
+
+    :type   to_date:    str (e.g., 2015-05-01)
+
+    :type   edit_permission: bool
+
+    :param  owner: (optional) - to get a list of resources owned by a specified username
+
+    :param  types: (optional) - to get a list of resources of the specified resource types
+
+    :param  from_date: (optional) - to get a list of resources created on or after this date
+
+    :param  to_date: (optional) - to get a list of resources created on or before this date
+
+    :param  edit_permission: (optional) - to get a list of resources for which the authorised user has edit permission
+
+    :rtype:  json string
+
+    :return:  a paginated list of resources with data for resource id, title, resource type, creator, sharing status,
+    date created, date last updated, resource bag url path, and science metadata url path
+
+    example return json format:
+
+        {   "count":n
+            "next": link to next page
+            "previous": link to previous page
+            "results":[
+                    {"resource_type": resource type, "resource_title": resource title, "resource_id": resource id,
+                    'creator': creator name, 'sharing_status': Public or Private, 'date_created': date resource created,
+                    'date_last_update': date resource last updated, 'bag': link to bag file, 'science_metadata':
+                    link to science metadata},
+                    {"resource_type": resource type, "resource_title": resource title, "resource_id": resource id,
+                    'creator': creator name, 'sharing_status': Public or Private, 'date_created': date resource created,
+                    'date_last_update': date resource last updated, 'bag': link to bag file, 'science_metadata':
+                    link to science metadata},
+            ]
+        }
+
+    :raises:
+
+    &nbsp;&nbsp;&nbsp;ValidationError: return json format: {'parameter-1':['error message-1'], 'parameter-2': ['error message-2'], .. }
+
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    pagination_class = PageNumberPagination
+
+    def get_queryset(self):
+        return self._get_resource_list()
```

Comment:
if all `get_queryset()` does is call `_get_resource_list()` then why not just implement all that as the body of `get_queryset()`